File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamQuaTangForm.ts
import { BooleanEditor, initFormType, IntegerEditor, PrefixedContext, ServiceLookupEditor } from "@serenity-is/corelib";

export interface SanPhamQuaTangForm {
    SanPhamId: ServiceLookupEditor;
    QuaTangId: IntegerEditor;
    ThuTu: IntegerEditor;
    IsDefault: BooleanEditor;
}

export class SanPhamQuaTangForm extends PrefixedContext {
    static readonly formKey = 'SanPhamDB.SanPhamQuaTang';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SanPhamQuaTangForm.init) {
            SanPhamQuaTangForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = IntegerEditor;
            var w2 = BooleanEditor;

            initFormType(SanPhamQuaTangForm, [
                'SanPhamId', w0,
                'QuaTangId', w1,
                'ThuTu', w1,
                'IsDefault', w2
            ]);
        }
    }
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamQuaTang/SanPhamQuaTangForm.cs
namespace wheystore.SanPhamDB.Forms;

[FormScript("SanPhamDB.SanPhamQuaTang")]
[BasedOnRow(typeof(SanPhamQuaTangRow), CheckNames = true)]
public class SanPhamQuaTangForm
{
    public int SanPhamId { get; set; }
    public int QuaTangId { get; set; }
    public int ThuTu { get; set; }
    public bool IsDefault { get; set; }
}